### ISSUE TITLE
Disk Size

### ISSUE THREAD

**Issue by capriciousduck** [labbots/google-drive-upload#118]
Is it possible to upload files that are bigger than my server disk? I have an instance on GCE(the free tier f1-micro VM). It has only a 30GB disk. Is there any possibility to upload files in chunks so that I can upload files that are bigger than 30Gb in total?

I am using this script on my other servers and I am satisfied. However, in this, I need to transfer bigger files and since traffic to google drive from a GCE VM is free, I want to use the free-tier VM for the transfer process. Any ideas or any other scripts available?

**Comment by Akianonymus** [comment]
Interesting idea. Is it possible ? Yes, google drive supoorts multi part uploading.

Reference: https://developers.google.com/drive/api/v3/manage-uploads#http---multiple-requests

About the source file, i am assuming you are going to download the file.

So, let's say, we download first 50 % of the file, e.g: 50 MB of total 100 MB file.

Then, we upload the 50 MB first, then download the rest 50% of the file and upload it.

That's the idea basically.

In short, you need following:

- Remote file should be partially downloadable
- Exact size of remote file in bytes ( which we can get by a curl head request )

I don't know if any existing script for it, most probably not. I would love to make one, when free.

Being said that, this is not in the scope of the script, so closing this issue. The conversation can be continued here.
